src/routes/jira/jira-get-connections-backfill-status.ts

```diff
@@ -0,0 +1,110 @@
+import { NextFunction, Request, Response } from "express";
+import { groupBy } from "lodash";
+import { RepoSyncState } from "~/src/models/reposyncstate";
+import { Subscription, SyncStatus } from "~/src/models/subscription";
+import {
+	mapSyncStatus,
+	ConnectionSyncStatus,
+	getRetryableFailedSyncErrors
+} from "~/src/util/github-installations-helper";
+
+type SubscriptionBackfillState = {
+	totalRepos?: number;
+	syncedRepos?: number;
+	syncStatus: ConnectionSyncStatus;
+	isSyncComplete: boolean;
+	backfillSince?: string;
+	failedSyncErrors?: Record<string, number>;
+	syncWarning?: string;
+};
+
+type BackFillType = {
+	[key: string]: SubscriptionBackfillState;
+};
+
+export const JiraGetConnectionsBackfillStatus = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+): Promise<void> => {
+	try {
+		const { jiraHost: localJiraHost } = res.locals;
+		const subscriptionIds = String(req.query?.subscriptionIds)
+			.split(",")
+			.map(Number)
+			.filter(Boolean);
+
+		if (subscriptionIds.length === 0) {
+			req.log.warn("Missing Subscription IDs");
+			res.status(400).send("Missing Subscription IDs");
+			return;
+		}
+
+		const subscriptions = await Subscription.findAllForSubscriptionIds(subscriptionIds);
+
+		const resultSubscriptionIds = subscriptions.map(
+			(subscription) => subscription.id
+		);
+
+		if (!subscriptions || subscriptions.length === 0) {
+			req.log.error("Missing Subscription");
+			res.status(400).send("Missing Subscription");
+			return;
+		}
+
+		const jiraHosts = subscriptions.map(
+			(subscription) => subscription?.jiraHost
+		);
+
+		const jiraHostsMatched = jiraHosts.every(
```

Comment:
/nit 🤔 - i think you could drop the map and just use the every like this

const jiraHostsMatched = subscriptions.every(
    (subscription) => subscription?.jiraHost === localJiraHost
);

Reply:
yup thats r8 short and sweet! done